### PR TITLE
feat(shared-components): add standardized EmptyState component with tests and documentation

### LINKED ISSUE #5120: Phase 1: EmptyState Migration - Component Foundation
# Phase 1: EmptyState Migration - Component Foundation

1. **Part of:** #5087 - EmptyState Component Implementation
2. **PR Number:** 1 of 7
3. **Estimated Effort:** 6-8 hours
4. This is a detailed guide for the tasks to perform. There may be mistakes, so you'll need to use your discretion in tackling the various code changes and PRs. Use your initiative to have the best outcome for the long term maintainability of the code base.
6. **_IMPORTANT_:** Refer to this link for more detailed plans:
    - https://github.com/PalisadoesFoundation/talawa-admin/issues/5111#issuecomment-3671736597

## Overview

Create the foundational EmptyState component with TypeScript interfaces, implementation, tests, and documentation following established codebase patterns.

### New Files
- `src/shared-components/EmptyState/EmptyState.tsx`
- `src/shared-components/EmptyState/EmptyState.spec.tsx`
- `src/shared-components/EmptyState/EmptyStateMocks.ts`
- `src/types/shared-components/EmptyState/interface.ts`

### Modified Files
- `docs/docs/docs/developer-resources/reusable-components.md` (add EmptyState section)

## Acceptance Criteria

- [ ] TypeScript interface created with required `message` prop and optional `description`, `icon`, `action`, `className`, `dataTestId` props
- [ ] Component implemented using MUI Stack with centered alignment
- [ ] i18n integration with `useTranslation` hook and fallback to plain strings
- [ ] Accessibility attributes included (`role="alert"`, `aria-label`)
- [ ] Comprehensive TSDoc documentation with `@component`, `@category`, `@remarks`, `@example` tags
- [ ] Test coverage reaches 100% with tests for all prop combinations
- [ ] Mock data file created for testing
- [ ] Component added to reusable components documentation

### Interface (`src/types/shared-components/EmptyState/interface.ts`)

```typescript
/**
 * Props interface for the EmptyState component.
 * 
 * @interface InterfaceEmptyStateProps
 * @category Shared Components
 */
export interface InterfaceEmptyStateProps {
  /**
   * Primary message to display (i18n key or plain string).
   * @required
   */
  message: string;

  /**
   * Optional secondary description text.
   */
  description?: string;

  /**
   * Icon to display above the message.
   */
  icon?: string | React.ReactNode;

  /**
   * Action button configuration.
   */
  action?: {
    label: string;
    onClick: () => void;
    variant?: 'primary' | 'secondary' | 'outlined';
  };

  /**
   * Custom CSS class name.
   */
  className?: string;

  /**
   * Test identifier.
   */
  dataTestId?: string;
}
```

### Component Structure

```typescript
const EmptyState: React.FC<InterfaceEmptyStateProps> = ({
  message,
  description,
  icon,
  action,
  className,
  dataTestId = 'empty-state',
}) => {
  const { t } = useTranslation();
  
  return (
    <Stack
      height="100%"
      alignItems="center"
      justifyContent="center"
      spacing={2}
      padding={4}
      className={className}
      role="alert"
      aria-label={getText(message)}
      data-testid={dataTestId}
    >
      {/* Icon, Message, Description, Action sections */}
    </Stack>
  );
};
```

## Testing Requirements

- Basic rendering with message only
- Conditional rendering of icon, description, action
- i18n translation and fallback behavior
- Accessibility attributes verification
- Action button onClick handler
- Custom className application

## Links

- **Comprehensive Plan:** [Comment #3671736597](https://github.com/PalisadoesFoundation/talawa-admin/issues/5111#issuecomment-3671736597)
- **Parent Issue:** #5087

### PR DESCRIPTION
**Pull request by adityai0**
**What kind of change does this PR introduce?**

Feature – introduces a new reusable shared UI component.

---

**Issue Number:**

Fixes #5120 

---

**Snapshots/Videos:**

N/A – UI component validated via comprehensive unit tests.

---

**If relevant, did you update the documentation?**

Yes  
Documentation updated under **Reusable Components**:  
https://docs-admin.talawa.io/docs/developer-resources/reusable-components

---

## Summary

This PR introduces a standardized **EmptyState** reusable component to ensure consistent handling of empty or no-data states across the application.

### Motivation
Previously, empty states were implemented inconsistently across different screens, leading to duplicated logic, inconsistent UI, and harder maintenance. This change consolidates empty state handling into a single, well-defined shared component.

### What this PR includes
- A reusable `EmptyState` component with:
  - Required `message` prop
  - Optional `description`, `icon`, `action`, `className`, and `dataTestId`
- Built using MUI `Stack` with centered alignment
- i18n integration with graceful fallback to plain strings
- Built-in accessibility (`role="alert"`, `aria-label`)
- Comprehensive TSDoc documentation with examples
- Mock data file for cleaner and reusable tests
- Full test coverage (≥95%) including:
  - Rendering scenarios
  - Conditional props
  - Action button interaction
  - Accessibility checks
  - i18n fallback behavior
- Component documented in the reusable components guide

This results in cleaner UI logic, improved accessibility, easier testing, and ~70% reduction in duplicate empty-state code.

---

**Does this PR introduce a breaking change?**

No  
This is an additive change and does not affect existing components or APIs.

---

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

---

**Other information**

N/A

---

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **New Features**
  * Added an EmptyState component with customizable message, optional icon (name or custom element), optional description, optional action button, styling hook, built-in translation support and accessibility attributes.
  * Added a public props interface to configure the component.

* **Tests**
  * Added comprehensive unit tests covering rendering variants, action behavior and styling, accessibility attributes, data-test-id hooks, and i18n fallback handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by adityai0** [comment]
@coderabbit Please review for correctness, test coverage, and alignment with project standards. Flag any issues or improvements.

**Comment by adityai0** [comment]
@coderabbitai All highlighted issues have been addressed.  
Please re-evaluate this PR and flag any remaining issues, including minor or trivial ones.  
If everything looks good, please confirm that the PR is ready to merge.

**Comment by adityai0** [comment]
@coderabbitai All highlighted issues have been addressed.  
Accessibility updated using `role="status"` with `aria-live="polite"` and no redundant ARIA.  
Please re-review and flag any remaining issues, even minor ones. If none, confirm merge readiness.

File: src/shared-components/EmptyState/EmptyState.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import {
+  emptyStateBaseMock,
+  emptyStateWithDescriptionMock,
+  emptyStateWithIconMock,
+  emptyStateWithCustomIconMock,
+  emptyStateBaseForActionMock,
+  emptyStateWithAllPropsMock,
+  emptyStateWithCustomCSSMock,
+  emptyStateWithCustomDataTestIdMock,
+} from './EmptyStateMocks';
```
[the remark this answers is not in the document]
Comment (adityai0) [reply src/shared-components/EmptyState/EmptyState.spec.tsx:15]: Hey @coderabbitai , i double check ed it, its correct

File: src/shared-components/EmptyState/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+  return (
+    <Stack
+      height="100%"
+      alignItems="center"
+      justifyContent="center"
+      spacing={2}
+      padding={4}
+      role="status"
+      aria-live="polite"
```
[the remark this answers is not in the document]
Comment (adityai0) [reply src/shared-components/EmptyState/EmptyState.tsx:116]: Why aria-live="polite" is correct here?

- Announces content without interrupting
- Screen readers read it when idle
- No duplicate announcements
- No urgency implied

**Comment by adityai0** [comment]
Hi @palisadoes , CodeRabbit AI has completed its review and approved this PR.  
It’s ready for your review whenever convenient. Thank you!

**Comment by palisadoes** [comment]
@coderabbitai full review

Perform a comprehensive analysis and reevaluate all resolved items with the following criteria:

**Test Coverage Analysis:**
1. For each modified or new test file, verify:
   - All edge cases and use cases are covered (success paths, error paths, boundary conditions)
   - Test assertions are correct and meaningful
   - 100% code coverage is achieved for the associated implementation files
   - No redundant or duplicate test cases exist
   
2. For each modified or new source file, identify:
   - Specific lines of code lacking test coverage
   - Functions, branches, or conditions that need additional test cases
   - Mock/stub requirements that may be missing

**Code Quality Assessment:**
3. Identify any files that should be excluded:
   - Accidentally committed files (IDE configs, env files, temporary files)
   - Files unrelated to the PR scope
   - Generated files that should be in .gitignore

**Requirement Validation:**
4. Verify the PR fully addresses the linked issue:
   - All acceptance criteria from the issue are met
   - Implementation matches the described solution approach
   - No scope creep or unrelated changes are included

5. Confirm compliance with all previous review feedback:
   - All resolved comments have been properly addressed
   - No regressions of previously suggested improvements
   - Coding standards and conventions are followed

**Review Decision:**
- Request changes (CHANGES_REQUESTED) if ANY of the following apply:
  - Missing test coverage or invalid tests
  - Code quality issues, security concerns, or bugs
  - Incomplete implementation of issue requirements
  - Non-compliance with previous feedback
  - Presence of unnecessary files

- Approve (APPROVED) only if:
  - All criteria above are fully satisfied
  - No changes, even trivial ones, are required

**Comment by adityai0** [comment]
sure I'll be back with the requested changes asap

**Comment by adityai0** [comment]
@coderabbitai Added tests to cover remaining `getButtonVariant` switch branches.  
Coverage now reaches 100%. Kindly re-review and letme know if it is ready to merger?
 Thanks!

**Comment by adityai0** [comment]
@coderabbitai
Added a test to cover the default (`undefined`) variant branch in `getButtonVariant`.  
All switch branches are now covered and test coverage is 100%. Kindly re-review and let me know if it is ready to merge.

**Comment by adityai0** [comment]
Hi @palisadoes , CodeRabbit AI has completed its review and approved this PR.
It’s ready for your review whenever convenient. Thank you!

**Comment by palisadoes** [comment]
@coderabbitai full review using this prompt:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/5406#issuecomment-3682023572
